pkg/cable/wireguard/driver.go

```diff
@@ -111,15 +114,42 @@ func NewDriver(localSubnets []string, localEndpoint types.SubmarinerEndpoint) (c
 		return nil, fmt.Errorf("failed to configure WireGuard device: %v", err)
 	}
 
-	klog.V(log.DEBUG).Infof("Initialized WireGuard %s with publicKey %s", DefaultDeviceName, pub)
+	klog.V(log.DEBUG).Infof("Created WireGuard %s with publicKey %s", DefaultDeviceName, pub)
 	return &w, nil
 }
 
 func (w *wireguard) Init() error {
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
+
+	klog.V(log.DEBUG).Infof("Initializing WireGuard device for cluster %s", w.localEndpoint.Spec.ClusterID)
+
+	if len(w.connections) != 0 {
+		return fmt.Errorf("cannot initialize with existing connections: %+v", w.connections)
+	}
+	a, err := netlink.AddrList(w.link, unix.AF_INET)
+	if err != nil || len(a) == 0 {
+		return fmt.Errorf("cannot get WireGuard link address: %v", err)
+	}
+	d, err := w.client.Device(DefaultDeviceName)
+	if err != nil {
+		return fmt.Errorf("wgctrl cannot find WireGaurd device: %v", err)
+	}
+	k, err := keyFromSpec(&w.localEndpoint.Spec)
```

Comment:
A question just for my understanding: After wireguard successfully sets up the tunnels to remote clusters, if the Submariner Gateway Pod is restarted, will the tunnel be destroyed as soon as GatewayPod is terminated?

Reply:
- WG device remains, but UDP tunnels will timeout and fade away.
- The routes will be removed by RA -- it flushes table 220. 
- Once the other GW comes up, all peers will have the allowed IPs for this cluster pointing only to the new GW -- so no traffic from old GW would be accepted. 
- If a new gateway is restarted on the same node then WG device is destroyed and recreated.

(longer term it may make sense to create a pluggable fwk to call cable libraries for cleanup, by RA or perhaps operator)

Reply:
> * WG device remains, but UDP tunnels will timeout and fade away.
> * The routes will be removed by RA -- it flushes table 220.
> * Once the other GW comes up, all peers will have the allowed IPs for this cluster pointing only to the new GW -- so no traffic from old GW would be accepted.
> * If a new gateway is restarted on the same node then WG device is destroyed and recreated.

Okay, the behavior is same as IPsec for now where datapath disruption would happen. BTW, we are trying to avoid this with IPsec and I'm looking into it.
> 
> (longer term it may make sense to create a pluggable fwk to call cable libraries for cleanup, by RA or perhaps operator)

+1